modules/base/include/modules/base/algorithm/camerautils.h

```diff
@@ -0,0 +1,195 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/base/basemoduledefine.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/compositeproperty.h>
+#include <inviwo/core/properties/buttonproperty.h>
+#include <inviwo/core/ports/volumeport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/common/inviwo.h>
+
+#include <modules/base/algorithm/mesh/axisalignedboundingbox.h>
+
+namespace inviwo {
+
+namespace camerautil {
+
+enum class Side { XNegative, XPositive, YNegative, YPositive, ZNegative, ZPositive };
+
+/**
+ * Setup the camera parameters such that the whole bounding box spaned by basisAndOffset will be
+ * inside the view frustum.
+ *
+ * @param cam the Camera to update
+ * @param basisAndOffset the basis and offset of the bounding box that will fit inside the new view
+ * frustum
+ * @param side Which side of the bounding box that will face the camera
+ * @param fitRatio determines how much spacing will be between edge of view and edge of volume. A
+ * fit ratio of 1 means a perfect fit, not edge between view frustum and volume border (takes aspect
+ * ratio into acount)
+ * @param setNearFar set to true to also updated the cameras new/far clip ranges @see
+ * setCameraNearFar
+ * @param setLookRanges set to true to also updated the min/max ranges of the cameras
+ * lookTo/lookFrom ranges @see setCameraLookRanges
+ */
+IVW_MODULE_BASE_API void setCameraView(CameraProperty &cam, const mat4 &basisAndOffset, Side side,
+                                       float fitRatio = 1.05f, bool setNearFar = false,
+                                       bool setLookRanges = false);
+
+/**
+ * Set the ranges of the look to and look from properties of the camera. Will center around the mid
+ * point of dataToWorld. The lookTo will be set to ranges to stay within the volume. The lookFrom
+ * will be set to the mid point +- the basis vector of dataToWorld times the zoom factor. i.e a zoom
+ * factor of 25 will alow to zoom out to a distance of "25 volumes".
```

Comment:
"The lookFrom
 * will be set to the mid point +- the basis vector of dataToWorld times the zoom factor. That is a zoom
 * factor of 25 will allow to zoom out to a distance of "25 volumes".